vernac/attributes.ml

```diff
@@ -382,3 +382,27 @@ let typing_flags_parser : Declarations.typing_flags key_parser = fun orig args -
 
 let typing_flags =
   attribute_of_list ["bypass_check", typing_flags_parser]
+
+let modes_declaration_parser : Typeclasses.hint_mode list list key_parser = fun orig att ->
+  match att with
+  | VernacFlagLeaf (FlagString args) ->
+    let modes = CString.split_on_char ',' args in
+    let one_mode mode =
+      let args = CString.explode mode in
+      let parse_mode = function
+        | "-" -> Some Typeclasses.ModeOutput
+        | "+" -> Some Typeclasses.ModeInput
+        | "!" -> Some Typeclasses.ModeNoHeadEvar
+        | " " -> None
+        | s -> CErrors.user_err Pp.(str "Ill-formed “modes” specification: " ++ str s)
+      in CList.map_filter parse_mode args
+    in
+    let modes = List.map one_mode modes in
+    if CList.is_empty modes then
```

Comment:
The `modes` list cannot be empty because it is the mapped output of `split_on_char` (https://ocaml.org/api/String.html#VALsplit_on_char). So this is not the right test.